File: code/search/linear_search/linear_search.c
#include <stdio.h>
/*
 * Part of Cosmos by OpenGenus Foundation
*/

/*
 * Input: an integer array with size in index 0, the element to be searched
 * Output: if found, returns the index of the element else -1
*/
int search(int arr[], int size, int x)
{
	int i=0;
	for (i=0; i<size; i++)
		if (arr[i] == x)
		    return i;
	return -1;
}


int main() 
{
    int arr[] = {2,3,1,5}; // Index 0 stores the size of the array (initially 0)
    int size = sizeof(arr)/sizeof(arr[0]);
    int find = 1;
    printf("Position of %d is %d\n", find, search(arr,size,find));
    return 0;
}

File: code/search/src/linear_search/linear_search.js
/*
 * Part of Cosmos by OpenGenus Foundation
 */

/*
 * Input    :   An integer array and the element to be searched
 * Output   :   If found returns the index of element or else -1
 * Example  :
 *              var index = linearSearch([1, 2, 3, 4, 7, 8], 8);
 *
 */
function linearSearch(arr, element) {
  for (var i = 0; i < arr.length; i++) {
    if (arr[i] === element) {
      return i;
    }
  }
  return -1;
}
